sklearn/cluster/_k_means_common.pyx

```diff
@@ -97,10 +97,10 @@ def _euclidean_sparse_dense_wrapper(
 
 
 cpdef floating _inertia_dense(
-        np.ndarray[floating, ndim=2, mode='c'] X,  # IN
-        floating[::1] sample_weight,               # IN
-        floating[:, ::1] centers,                  # IN
-        int[::1] labels,                           # IN
+        floating[:, ::1] X,           # IN
+        floating[::1] sample_weight,  # IN
+        floating[:, ::1] centers,     # IN
+        int[::1] labels,              # IN
         int n_threads):
```

Comment:
Is this is occasion to mark some interfaces it as `nogil` such as this one?
```suggestion
cpdef floating _inertia_dense(
        floating[:, ::1] X,           # IN
        floating[::1] sample_weight,  # IN
        floating[:, ::1] centers,     # IN
        int[::1] labels,              # IN
        int n_threads) nogil:
```

Reply:
The loop is with `prange(...nogil=True)`, so it is not necessary. And AFAIK, only C function (`cdef`) can be marked with `nogil`.